### PR TITLE
IP: fixing Ethan's review (10.23)

### PR DESCRIPTION
**Pull request by vb-balazs**

### CODE REVIEW

File: ip/draft-ietf-detnet-ip.xml [hunk 1]
```diff
@@ -111,22 +111,28 @@
...
-        loss, assured latency, and limited out-of-order delivery). Since no DetNet
-        specific headers are added to support DetNet IP flows, only the
-        forwarding sub-layer functions are supported using the DetNet IP
-        defined by this document. Service protection can be provided on a per
+        loss, assured latency, and limited out-of-order delivery). 
+		The service sub-layer requires additional fields to provide its service; 
+		for example to provide service protection via PREOF, each packet must 
+		be assigned a sequence number. Since no DetNet-specific fields 
+		(such as sequence number) are added to support DetNet IP flows, 
+		the DetNet service sub-layer for IP cannot support PREOF. Thus only
```
Comment (louberger) [inline ip/draft-ietf-detnet-ip.xml]: These PREOF is a detail that doesn't belong in the IP draft -- e.g., if someone defines a different protection mechanism it shouldn't impact this doc! How about:
The service sub-layer generally requires additional fields to provide its service; for example  see <xref target="I-D.ietf-detnet-data-plane-framework"/>. Since no DetNet-specific fields 
are added to support DetNet IP flows, only

File: ip/draft-ietf-detnet-ip.xml [hunk 2]
```diff
@@ -277,14 +284,13 @@
...
-        traffic that is identified as DetNet flows, relay nodes understand the
+        traffic those are identified within the DetNet domain as DetNet flows, 
+		relay nodes understand the
         forwarding requirements of the DetNet flow and ensure that node,
         interface and sub-network resources are allocated to ensure DetNet
         service requirements. The dotted line around the Service component of
         the Relay Nodes indicates that the transit routers are DetNet service
         aware but do not perform any DetNet service sub-layer function, e.g., PREOF.
-        IEEE 802.1 TSN is an example sub-network type which can provide support
-        for DetNet flows and service.
```
Comment (louberger) [inline ip/draft-ietf-detnet-ip.xml:287]: What is the value of deleting this informative sentence?

File: ip/draft-ietf-detnet-ip.xml [hunk 3]
```diff
@@ -357,46 +363,48 @@
...
         <t>
-          Data-flows requiring DetNet service are generated and terminated on
-          end systems.  This document deals only with IP end systems.  The
-          protocols used by an IP end system are specific to an application
-          and end systems peer with end systems using the same application
-          encapsulation format.  This said, DetNet's use of 6-tuple IP flow
+          IP Data-flows requiring DetNet service are generated and terminated 
+		  on IP end systems.
+		  Encapsulation of application data on the IP end system is application 
+		  specific and it has to match the encapsulation format of the peer IP
```
Comment (louberger) [inline ip/draft-ietf-detnet-ip.xml]: Who is "it" in this sentence?  Also this document can't place requirements on non-detnet components.  If the change is being made for editorial reasons, I suggest using the original text -- if not, let's discuss on list (perhaps as part of last call.

File: ip/draft-ietf-detnet-ip.xml [hunk 4]
```diff
@@ -546,9 +555,9 @@ DetNet                    |L2/SbN|          |L2/SbN|
           One additional consideration for DetNet nodes which support CoS
           services is that they MUST ensure that the CoS service classes do
           not impact the congestion protection and latency control mechanisms
-          used to provide DetNet QoS.  This requirement is similar to
-          requirement for MPLS LSRs to that CoS LSPs do not impact the
-          resources allocated to TE LSPs via <xref target="RFC3473"/>.
+          used to provide DetNet QoS.  This requirement is similar to the 
+		  requirement for MPLS LSRs that CoS LSPs must not impact the resources
```
Comment (louberger) [inline ip/draft-ietf-detnet-ip.xml]: It's generally good practice to avoid lower cased conformance language wherever possible:
s/must/cannot

File: ip/draft-ietf-detnet-ip.xml [hunk 5]
```diff
@@ -569,7 +578,7 @@ DetNet                    |L2/SbN|          |L2/SbN|
           but that have not been the subject of a completed reservation,
           can disrupt the QoS offered to properly reserved DetNet flows
           by using resources allocated to the reserved flows.
-          Therefore, the network nodes of a DetNet network must:
+          Therefore, the network nodes of a DetNet network MUST:
```
Comment (louberger) [inline ip/draft-ietf-detnet-ip.xml]: I think there's an open issue on this text that I thought I fixed a while back -- will submit under a separate PR

File: ip/draft-ietf-detnet-ip.xml [hunk 6]
```diff
@@ -871,7 +880,7 @@ DetNet                    |L2/SbN|          |L2/SbN|
           Specifically, implementations of this document SHALL use
           management and control information to select the one or more
           outgoing interfaces and next hops to be used for a packet
-          belonging to a DetNet flow.
+          assiciated to a DetNet flow.
```
Comment (louberger) [inline ip/draft-ietf-detnet-ip.xml]: spelling: associated with

**Review by louberger — CHANGES_REQUESTED** [review]
some changes  impacted intent and think should be rejected and/or discussed on list

File: ip/draft-ietf-detnet-ip.xml (hunk 1 above)
Comment (vb-balazs) [reply ip/draft-ietf-detnet-ip.xml]: Yes, text is intended here for explanation, why no PREOF for IP. Is it disturbing? Do You propose to delete? If yes, which part?

File: ip/draft-ietf-detnet-ip.xml (hunk 2 above)
Comment (vb-balazs) [reply ip/draft-ietf-detnet-ip.xml:287]: It is redundant. In the next sentence we list various sub-network technologies including TSN.

File: ip/draft-ietf-detnet-ip.xml (hunk 3 above)
Comment (vb-balazs) [reply ip/draft-ietf-detnet-ip.xml]: This sentence was intended as editorial clarification based on Ethan's comment. I have created this sentence, but as You commented it seems not to provide the required clarification. The original was not clear enough as well. I would suggest to solve this during in the last call. Alternatively please propose a new sentence with better phrasing.

File: ip/draft-ietf-detnet-ip.xml (hunk 4 above)
Comment (vb-balazs) [reply ip/draft-ietf-detnet-ip.xml]: OK. I have updated this.

File: ip/draft-ietf-detnet-ip.xml (hunk 5 above)
Comment (vb-balazs) [reply ip/draft-ietf-detnet-ip.xml]: OK.

File: ip/draft-ietf-detnet-ip.xml (hunk 6 above)
Comment (vb-balazs) [reply ip/draft-ietf-detnet-ip.xml]: OK. I have updated.

File: ip/draft-ietf-detnet-ip.xml (hunk 1 above)
Comment (louberger) [reply ip/draft-ietf-detnet-ip.xml]: As mentioned on the list, this document shouldn't depend on a reliability approach used in other documents or need to change when new reliability mechanisms are introduced.  I'm proposing:
OLD:
		The service sub-layer requires additional fields to provide its service; 
		for example to provide service protection via PREOF, each packet must 
		be assigned a sequence number. Since no DetNet-specific fields 
		(such as sequence number) are added to support DetNet IP flows, 
		the DetNet service sub-layer for IP cannot support PREOF. Thus only 
NEW
The service sub-layer generally requires additional fields to provide its service; for example see [DETNET-MPLS]. Since no DetNet-specific fields are added to support DetNet IP flows, only

File: ip/draft-ietf-detnet-ip.xml (hunk 2 above)
Comment (louberger) [reply ip/draft-ietf-detnet-ip.xml:287]: okay

File: ip/draft-ietf-detnet-ip.xml (hunk 1 above)
Comment (vb-balazs) [reply ip/draft-ietf-detnet-ip.xml]: OK. I have updated this.

**Review by louberger — APPROVED** [review]
As discussed in Nov 28 call
